### DISCUSSION TITLE
Best approach to check or uncheck other checkboxes in form.

### DISCUSSION THREAD

**Discussion by technoY2K** [formkit/formkit#1272] [General]
Hello, wanted to ask for some guidance as to the best way to uncheck checkboxes dependent on a single one. For example, I have checkboxes in different sections, some are multi-checkboxes (Arrays) and others are single checkboxes (Booleans). When a user checks the `unsubscribe` checkbox, if its true, then I want to uncheck all the other checkboxes if they are checked. I'm trying to figure out the best way to approach this, but don't see a clear path in how to update the forms value programmatically after a check event. The following is an example of the structure I currently have. With the following code, how could I update the checkbox values? Should I use sections-schema, event listeners on each input, etc?

```
<FormKit
  id="form"
  type="form"
  :actions="false"
  >
  <FormKit type="checkbox" label="..." name="..." />
  <FormKit type="checkbox" label="..." name="..." />
  <FormKit type="checkbox" label="..." name="..." />
  <FormKit type="checkbox" label="..." name="..." :option=[...] />
  <FormKit type="checkbox" label="..." name="..." :option=[...] />
</FormKit>
```

**Comment by fenilli** [comment]
Set the value of each one for `[]` or `undefined`.

**Reply by technoY2K** [reply]
What would be the approach to "listen" to when a specific checkbox has been checked? I was thinking about using a [plugin](https://formkit.com/essentials/architecture#plugins). Is that the most straightforward way?

**Reply by fenilli** [reply]
You can listen using the `node.on('commit', ())`, I would say yes a plugin would be the most straightforward way to deal with it, you can make the plugin receive the names of the checkboxes it will control, or even make a group that has all checkboxes instead, and target that for value.

**Reply by technoY2K** [reply]
Ok, ty. I'll try that.
